lib/util.js

```diff
@@ -33,6 +40,112 @@ util = {
         return url.unparse(urlObj);
     },
 
+    /**
+     * A generic utility method to sanitize variable transformations across collection formats.
+     *
+     * @param entity {Object} - A generic object that could contain variable data.
+     * @param override {Object} - A set of optional overrides for variable transformations.
+     * @return {Array} - The set of sanitized entity level variables.
+     */
+    handleVars: function (entity, override) {
+        var source = entity && (entity.variables || entity.variable),
+            result = _.map((override && override.values) || source, function (item) {
```

Comment:
`override && override.values) || source` this could lead to loss of collection level variables

Reply:
Will reverse the precedence for now.